hubtools/src/caboose.rs

```diff
@@ -193,10 +224,38 @@ mod tests {
             .board("bar")
             .name("fizz")
             .version("buzz")
+            .epoch("")
             .build();
         assert_eq!(caboose.git_commit(), Ok("foo".as_bytes()));
         assert_eq!(caboose.board(), Ok("bar".as_bytes()));
         assert_eq!(caboose.name(), Ok("fizz".as_bytes()));
         assert_eq!(caboose.version(), Ok("buzz".as_bytes()));
+        assert_eq!(caboose.epoch(), Ok("0".as_bytes()));
+    }
+
+    #[test]
+    fn builder_can_make_caboose_with_non_zero_epoch() {
+        let caboose = CabooseBuilder::default().epoch("1234567890").build();
+        assert_eq!(caboose.epoch(), Ok("1234567890".as_bytes()));
+    }
+
+    #[test]
+    fn builder_will_normalize_empty_epoch() {
+        let caboose = CabooseBuilder::default().epoch("").build();
+        assert_eq!(caboose.epoch(), Ok("0".as_bytes()));
+    }
+
+    #[test]
+    fn builder_will_normalize_short_epoch() {
+        let caboose = CabooseBuilder::default().epoch("1234").build();
+        assert_eq!(caboose.epoch(), Ok("1234".as_bytes()));
+    }
+
+    #[test]
+    fn builder_will_convert_u64_epoch_to_zero() {
+        let caboose = CabooseBuilder::default()
+            .epoch("4294967296") // u32::MAX_VALUE + 1
+            .build();
```

Comment:
If someone provides an invalid epoch, why should we normalize it to 0?

(This is a moot point if you change the type of `epoch(..)` to take a `u32`, which would make it impossible to call with an invalid input)

Reply:
It has been mootified.